drivers/mcp2515/candev_mcp2515.c

```diff
@@ -0,0 +1,582 @@
+        /* same mask */
+        if (dev_mcp->masks[mailbox_index] == f.can_mask) {
+            int filter_pos = 0;
+            /* stop at the end of mailbox or filter_id found */
+            while (filter_pos < _max_filters(mailbox_index) &&
+                   dev_mcp->filter_ids[mailbox_index][filter_pos] != f.can_id) {
+                filter_pos++;
+            }
+
+            /* filter id found */
+            if (filter_pos < _max_filters(mailbox_index)) {
+                /* remove filter */
+                mcp2515_set_filter(dev_mcp,
+                                   MCP2515_FILTERS_MB0 * mailbox_index + filter_pos,
+                                   CAN_EFF_MASK);
+                /* save modification */
+                dev_mcp->filter_ids[mailbox_index][filter_pos] = 0;
+
+                /* check mailbox empty */
+                int nb_item = 0;
+                for (int i = 0; i < _max_filters(mailbox_index); i++) {
+                    if (dev_mcp->filter_ids[mailbox_index][i] == 0) {
+                        nb_item++;
+                    }
+                }
+
+                /* mailbox empty */
+                if (nb_item == _max_filters(mailbox_index)) {
+                    /* remove mask */
+                    mcp2515_set_mask(dev_mcp, mailbox_index, CAN_EFF_MASK);
+                    /* save modification */
+                    dev_mcp->masks[mailbox_index] = 0;
+                }
+
+                filter_removed = 0;
+            }
+        }
+        mailbox_index++;
+    }
+    mcp2515_set_mode(dev_mcp, mode);
+
+    return filter_removed;
+}
+
+static void _irq_rx(candev_mcp2515_t *dev, int box)
+{
+    DEBUG("Inside mcp2515 rx irq, box=%d\n", box);
+
+    mcp2515_receive(dev, &dev->rx_buf[box], box);
+
+    _send_event(dev, CANDEV_EVENT_RX_INDICATION, &dev->rx_buf[box]);
+}
+
+static void _irq_tx(candev_mcp2515_t *dev, int box)
+{
+    DEBUG("Inside mcp2515 tx irq\n");
+    const struct can_frame *frame = dev->tx_mailbox[box];
+    dev->tx_mailbox[box] = NULL;
+
+    _send_event(dev, CANDEV_EVENT_TX_CONFIRMATION, (void *)frame);
+}
+
+static void _irq_error(candev_mcp2515_t *dev)
+{
+    uint8_t err;
+
+    DEBUG("Inside mcp2515 error irq\n");
+
+    err = mcp2515_get_errors(dev);
+
+    if (err & (ERR_WARNING | ERR_RX_WARNING | ERR_TX_WARNING)) {
+        DEBUG("Error Warning\n");
+        _send_event(dev, CANDEV_EVENT_ERROR_WARNING, NULL);
+    }
+    else if (err & (ERR_RX_PASSIVE | ERR_TX_PASSIVE)) {
+        DEBUG("Error Passive\n");
+        _send_event(dev, CANDEV_EVENT_ERROR_PASSIVE, NULL);
+    }
+    else if (err & ERR_TX_BUS_OFF) {
+        DEBUG("Bus Off\n");
+        _send_event(dev, CANDEV_EVENT_BUS_OFF, NULL);
+    }
+    else if (err & (ERR_RX_0_OVERFLOW | ERR_RX_1_OVERFLOW)) {
+        DEBUG("RX overflow\n");
+        _send_event(dev, CANDEV_EVENT_RX_ERROR, NULL);
+    }
+}
+
+static void _irq_message_error(candev_mcp2515_t *dev)
+{
+    (void)dev;
+#if (0)
```

Comment:
I think this was originally removed when the can_frame struct was changed. 
I adapted it to work with the new code, but had to remove a bunch of stuff. It may still need some work to 100% do what it's supposed to do.